### PR TITLE
Sso login branch

### PR DESCRIPTION
**Pull request by kdrtibor**
This branch is based on ssoPreparation started by Radislav, moving forward from this strategy I implemented the AuthenticationStrategy interface as well as added some extra methods to the interface.

### CODE REVIEW

File: OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs [hunk 1]
```diff
@@ -18,14 +31,14 @@ public class LwssoAuthenticationStrategy : AuthenticationStrategy
         private static string LWSSO_COOKIE_NAME = "LWSSO_COOKIE_KEY";
         private static string OCTANE_USER_COOKIE_NAME = "OCTANE_USER";
 
-        private ConnectionInfo credentials;
+        private UserPassConnectionInfo credentials;
```
Comment (radislavB) [inline OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs]: credentials might be also APIKeyConnectionInfo. Please revert it to ConnectionInfo

File: OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs [hunk 2]
```diff
@@ -18,14 +31,14 @@ public class LwssoAuthenticationStrategy : AuthenticationStrategy
...
         private string lwSsoCookie;
         private string octaneUserCookie;
         private String host;
 
         public LwssoAuthenticationStrategy(ConnectionInfo credentials)
         {
-            this.credentials = credentials;
+            this.credentials = credentials as UserPassConnectionInfo;
```
Comment (radislavB) [inline OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs]: see previous comment

File: OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs [hunk 3]
```diff
@@ -164,5 +177,22 @@ public void PrepareRequest(HttpWebRequest request)
             request.CookieContainer.Add(new Cookie(LWSSO_COOKIE_NAME, lwSsoCookie, cookiePath, cookieDomain));
             request.CookieContainer.Add(new Cookie(OCTANE_USER_COOKIE_NAME, octaneUserCookie, cookiePath, cookieDomain));
         }
+
+        public Task<String> GetWorkspaceUser()
+        {
+            return Task.FromResult(credentials.user);
```
Comment (radislavB) [inline OctaneSdk/Connector/Authentication/LwssoAuthenticationStrategy.cs]: here the most simple way is to check real implementation of connection info and return user for UserPassConnectionInfo or client_id for APIKeyConnectionInfo

File: OctaneSdk/Connector/Credentials/APIKeyConnectionInfo.cs [hunk 4]
```diff
@@ -25,18 +25,14 @@ namespace MicroFocus.Adm.Octane.Api.Core.Connector
 	/// </summary>
 	public class APIKeyConnectionInfo : ConnectionInfo
     {
+        public string access_token { get; set; }
```
Comment (radislavB) [inline OctaneSdk/Connector/Credentials/APIKeyConnectionInfo.cs]: Please explain why you need to enrich this class with additional properties.
Please consider creation new class with additional properties.

File: OctaneSdk/Entities/Users/WorkspaceUser.cs [hunk 5]
```diff
@@ -23,6 +23,6 @@ namespace MicroFocus.Adm.Octane.Api.Core.Entities
 	/// </summary>
 	public class WorkspaceUser : BaseUserEntity
     {
-        
+        public bool is_api_key { get; set; }
```
Comment (radislavB) [inline OctaneSdk/Entities/Users/WorkspaceUser.cs]: Generally we used to expose fields in another way , for example 
			{
				return GetStringValue(LANGUAGE_FIELD);
			}
			set
			{
				SetValue(LANGUAGE_FIELD, value);
			}

**Review by radislavB — CHANGES_REQUESTED** [review]
please review my comments

**Comment by kdrtibor** [comment]
I think I misunderstood what does the APIKeyConnectionInfo stands for and used it as connection info for sso type connection. I created the SsoConnectionInfo class and reverted the APIKeyConnectionInfo class.

**Review by radislavB — APPROVED** [review]
approved

**Pull request merged**